test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,167 @@
+});
+
+import { GeneratorOptions as GeneratorOptionsSDK, generatorOptionsCli as generatorOptionsSDK } from '@sap/cloud-sdk-generator';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import GenerateVdm, { FlagsParsed } from '../src/commands/generate-vdm';
+import * as foo from '../src/utils/generate-vdm-util';
+
+const spyToGeneratorSDK = jest.spyOn(foo, 'toGeneratorSDK');
+
+describe('generate-vdm', () => {
+  const pathForTests = path.resolve(__dirname, __filename.replace(/\./g, '-')).replace('-ts', '');
+
+  beforeAll(() => {
+    const pathForResources = path.resolve(__dirname, 'resources', 'template-generator-vdm');
+    fs.copySync(pathForResources, pathForTests);
+  });
+
+  afterAll(() => {
+    fs.removeSync(pathForTests);
+  });
+
+  afterEach(() => {
+    spyToGeneratorSDK.mockClear();
+  });
+
+  beforeEach(() => {
+    spyToGeneratorSDK.mockClear();
+  });
+
+  it('should generate a vdm.', async () => {
+    await GenerateVdm.run(['-i', 'edmxSource', '-o', 'generatedVdm', '--forceOverwrite', '--projectDir', pathForTests]);
+    const files = fs.readdirSync(path.join(pathForTests, 'generatedVdm', 'business-partner-service'));
+    expect(files.length).toBeGreaterThan(0);
+  }, 60000);
+
+  it('should fail if the mandatory parameters are not there', async () => {
+    try {
+      await GenerateVdm.run([]);
+    } catch (err) {
+      const messageNoFormatting = err.message as string;
+      expect(messageNoFormatting).toContain('-i, --inputDir INPUTDIR');
+    }
+  });
+
+  it('should parse strings and with --no- it should lead to all false.', async () => {
+    const expected = getAllFalse();
+    delete expected.projectDir;
+    try {
+      await GenerateVdm.run([...getInputAllFalse()]);
+    } catch (e) {
+      expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+    }
+  });
+
+  it('should resolve the projectdir', async () => {
+    const projectDir = '/root';
+    const expected = getDefault(projectDir);
+
+    try {
+      await GenerateVdm.run(['-i', 'input', '-o', 'output', '--projectDir', projectDir]);
+    } catch (e) {
+      expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+    }
+  });
+
+  it('should pass each boolean flags correctly', async () => {
+    return Object.keys(generatorOptionsSDK).map(async key => {
+      const option = generatorOptionsSDK[key as keyof GeneratorOptionsSDK];
+      const expected = getDefault('/somePrefix');
+      const args = ['-i', 'input', '-o', 'output', '--projectDir', '/somePrefix'];
+
+      if (option !== undefined && option.type === 'boolean') {
+        const casted = key as keyof GeneratorOptionsSDK;
+        if (!option.default) {
+          args.push(`--${key}`);
+          (expected[casted] as any) = true;
+        } else {
+          args.push(`--no-${key}`);
+          (expected[casted] as any) = false;
+        }
+        const spyToGeneratorSDK1 = jest.spyOn(foo, 'toGeneratorSDK');
+        try {
+          await GenerateVdm.run(args);
```

Comment:
Right now GenerateVdm.run is executed for every single bool param. I dont think this is intentional.

I assume you want to do this:
``` ts
  it('should pass each boolean flags correctly', async () => {
    const expected: { [key: string]: any } = getDefault('/somePrefix');
    const args = [
      '-i', 'input', 
      '-o', 'output', 
      '--projectDir', '/somePrefix', 
      ...Object.keys(generatorOptionsSDK).filter(key => {
        return typeof generatorOptionsSDK[key as keyof GeneratorOptionsSDK] === 'boolean';
      }).map(key => {
        const casted = key as keyof GeneratorOptionsSDK;
        if (!option.default) {
          expected[casted] = true;
          return `--${key}`;
        } else {
          expected[casted] = false;
          return `--no-${key}`;
        }
    })];

    const spyToGeneratorSDK1 = jest.spyOn(foo, 'toGeneratorSDK');
    try {
      await GenerateVdm.run(args);
    } catch (e) {
      expect(spyToGeneratorSDK1).toHaveReturnedWith(expected);
    }
  });
```